src/adapter/src/coord/command_handler.rs

```diff
@@ -365,6 +365,28 @@ impl Coordinator {
                     let _ = tx.send(result);
                 }
 
+                Command::CopyToPreflight {
+                    s3_sink_connection,
+                    sink_id,
+                    tx,
+                } => {
+                    // Spawn a background task to perform the slow S3 preflight operations.
+                    // This avoids blocking the coordinator's main task.
+                    let connection_context = self.connection_context().clone();
+                    task::spawn(|| "copy_to_preflight", async move {
+                        let result = mz_storage_types::sinks::s3_oneshot_sink::preflight(
+                            connection_context,
+                            &s3_sink_connection.aws_connection,
+                            &s3_sink_connection.upload_info,
+                            s3_sink_connection.connection_id,
+                            sink_id,
+                        )
+                        .await
+                        .map_err(AdapterError::from);
+                        let _ = tx.send(result);
+                    });
+                }
```

Comment:
I'm wondering if the coordinator needs to be involved here at all. `StorageConfiguration::connection_context` has this documentation:

```
    /// Immutable, CLI-configured parameters.
    ///
    /// TODO(guswynn): `ConnectionContext` also contains some shared global state that should
    /// eventually be moved up to this struct.
```

Which might mean that the `ConnectionContext` would be easily cacheable in the frontend task. It's not entirely clear because of the TODO that doesn't specify what the shared global state is supposed to be.

Reply:
Yeah, I'm 90% that it would be easy to do that, but I didn't want to take the remaining 10% risk now, because there were just too many things in `ConnectionContext` that I don't know much about. But if you feel we could just trust the `Should be kept cheaply cloneable` comment on `ConnectionContext`, then I'm happy to do the refactoring.